dd-java-agent/instrumentation/kafka-clients-0.11/src/main/java/datadog/trace/instrumentation/kafka_clients/TracingIterator.java

```diff
@@ -58,9 +59,14 @@ public boolean hasNext() {
 
   protected void decorate(ConsumerRecord<?, ?> val) {
     try {
+      final AgentSpan span;
       if (val != null) {
-        final Context spanContext = propagate().extract(val.headers(), GETTER);
-        final AgentSpan span = startSpan(operationName, spanContext);
+        if (!Config.get().getKafkaClientPropagationDisabledList().contains(val.topic())) {
+          final Context spanContext = propagate().extract(val.headers(), GETTER);
+          span = startSpan(operationName, spanContext);
+        } else {
+          span = startSpan(operationName);
```

Comment:
```suggestion
          span = startSpan(operationName, null);
```

I think we'll want the parent to be `null` here to ensure it doesn't inadvertently pick up an existing trace as parent.